packages/core/src/index.ts

```diff
@@ -122,31 +131,37 @@ export default class Engine<Name extends string = string> {
 				this.publicParsedRules[name] = rule as RuleNode<Name>
 			}
 		}
+
+		this.publicSituation = {} as Situation<Name>
 	}
 
 	resetCache() {
 		this.cache = emptyCache()
 	}
 
 	setSituation(
-		situation: Partial<Record<Name, PublicodesExpression | ASTNode>> = {},
-		options: { keepPreviousSituation?: boolean } = {},
+		situation: Situation<Name> = {},
+		options: {
+			keepPreviousSituation?: boolean
+			shouldFilterSituation?: boolean
+		} = {},
 	) {
 		this.resetCache()
 
 		const keepPreviousSituation = options.keepPreviousSituation ?? false
+		const shouldFilterSituation =
+			(options.shouldFilterSituation || this.baseContext.useSafeGetSituation) ??
+			false
 
-		Object.keys(situation).forEach((name) => {
-			if (!(name in this.baseContext.parsedRules)) {
-				throw new PublicodesError(
-					'EvaluationError',
-					`Erreur lors de la mise à jour de la situation : ${name} n'existe pas dans la base de règle.`,
-					{ dottedName: name },
-				)
-			}
+		const situationCopy = this.safeGetSituation({
```

Comment:
Si l'option `keepPreviousSituation` est activée, alors ce devrait être : 
```suggestion
		const situationCopy = {...this.publicSituation,...this.safeGetSituation({
```

Reply:
On pourrait d'ailleurs mettre directement à jour `this.publicSituation` ici, au lieu de créer une nouvelle variable

Reply:
J'avais compris qu'on ne souhaitait pas modifier directement la situation courante dans la première review ?